tests/ti_deps/deps/test_trigger_rule_dep.py

```diff
@@ -275,6 +275,44 @@ def test_all_done_tr_failure(self):
         self.assertEqual(len(dep_statuses), 1)
         self.assertFalse(dep_statuses[0].passed)
 
+    def test_none_skipped_tr_success(self):
+        """
+        None-skipped trigger rule success
+        """
+        ti = self._get_task_instance(TriggerRule.NONE_SKIPPED,
+                                     upstream_task_ids=["FakeTaskID",
+                                                        "OtherFakeTaskID",
+                                                        "FailedFakeTaskID"])
+        dep_statuses = tuple(TriggerRuleDep()._evaluate_trigger_rule(
+            ti=ti,
+            successes=2,
+            skipped=0,
+            failed=1,
+            upstream_failed=0,
+            done=3,
+            flag_upstream_failed=False,
+            session="Fake Session"))
+        self.assertEqual(len(dep_statuses), 0)
+
+    def test_none_skipped_tr_failure(self):
+        """
+        None-skipped trigger rule failure
+        """
+        ti = self._get_task_instance(TriggerRule.NONE_SKIPPED,
+                                     upstream_task_ids=["FakeTaskID",
+                                                        "SkippedTaskID"])
+        dep_statuses = tuple(TriggerRuleDep()._evaluate_trigger_rule(
+            ti=ti,
+            successes=1,
+            skipped=1,
+            failed=0,
+            upstream_failed=0,
+            done=2,
+            flag_upstream_failed=False,
+            session="Fake Session"))
+        self.assertEqual(len(dep_statuses), 1)
+        self.assertFalse(dep_statuses[0].passed)
```

Comment:
Could you add two more cases covering when `flag_upstream_failed` is true please? Happy for them to be extra asserts in the existing functions.